packages/nimble-components/src/select/specs/option-groups-hld.md

```diff
@@ -0,0 +1,122 @@
+
+```ts
+public class ListOptionGroup {
+    @attr
+    public label?: string;
+}
+```
+
+In addition to the `label` attribute, the `ListOptionGroup` will also support slotted labels. This will result in a configuration that looks like the following:
+
+```html
+<nimble-list-option-group>
+    Swedish Cars
+    <nimble-list-option value="volvo">Volvo</option>
+    <nimble-list-option value="saab">Saab</option>
+</nimble-list-option-group>
+```
+
+#### Usage notes
+
+1. If a client configures both the `label` attribute _and_ supplies slotted content, we will honor the label attribute.
+2. Expected slotted content structure will be documented in Storybook. An example of not specifically supported capability (and thus not documented) is arbitrary content, such as text mixed with icons.
+
+### Filtering
+
+As [mentioned in the IxD](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/select/specs/IxD.md#groups), when a user has filtering enabled, typing into the filter input will match both individual options (as normal), as well as _all_ options that are under a group that matches the filter text.
+
+When all options under a group are filtered out, the group will also be hidden in the dropdown.
+
+#### Combobox Guidance
+
+It should be noted that when using groups with a `Combobox`, it is advised that each option still be unique (for both its display and value). This is predicated on the [decision](https://github.com/ni/nimble/issues/1168#issuecomment-1568733159) that the `Combobox` will change its behavior for platforms like Angular, where the model value will _always_ be the text that is in the display. This will make it impossible to disambiguate between options that have the same display value under separate groups.
+
+### Accessibility
+
+-   The `ListOptionGroup` element will have a [role of "group"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role), which matches the role given to the native `optgroup` element.
+-   The `ListOptionGroup` should apply the `aria-label` attribute with a value that is the display text for the group.
+-   The `ListOptionGroup` element is ignored during keyboard navigation of the dropdown.
+-   Screen readers should report the group when the user navigates to an option within a group for the first time. This is achieved by ensuring the accessibility tree has a structure like the following:
+
+![](./spec-images/option-group-accessibility-tree.PNG)
+
+### Angular/Blazor Integration
+
+Standard implementation pattern for support.
+
+## Alternative Implementations / Designs
+
+Originally, there was a desire to display the group name for the selected option outside of the dropdown of the component. However, there is not precedent for this for other design systems that support groups, nor the native `select`. If a client desires to display the group for the selected option that will be managed on the client side, including _where_ the display would be.
```

Comment:
Double checking, are we satisfied that we provide good enough APIs for a client to implement this? I imagine it would look something like:
1. subscribe to the `change` event on the `select`
2. when it fires, use the event data or the value to find the `list-option` corresponding to the new value
3. navigate up the DOM of that option to find its group
4. get the title of that group either via `label` or `textContent`, depending on which one you set in your app

An alternative would be to add event data or helper methods to do some of this for the client. Probably low ROI, but worth considering.